src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/InversionBreakendPreFilter.java

```diff
@@ -0,0 +1,440 @@
+                        .map(Integer::valueOf).anyMatch(mq -> mq >= primitiveContigMQFilter);
+        if ( ! hasGoodMappings ) {
+            targetContainerFilteredAwayDueToMQ.add(invBreakEndContext);
+        } else { // filter using MQ
+            final SimpleInterval mateRefLoc = invBreakEndContext.getMateRefLoc();
+            final SVInterval svInterval = new SVInterval(refDict.getSequenceIndex(invBreakEndContext.getContig()),
+                    invBreakEndContext.getStart() - 1, mateRefLoc.getEnd());
+            if (svInterval.getLength() > mateDistanceFilterThreshold)
+                targetContainerFilteredAwayDueToSize.add(invBreakEndContext); // filter using size
+            else {
+                if (complexVariantSVIntervalTree.hasOverlapper(svInterval)) {
+                    final StringBuilder stringBuilder = new StringBuilder();
+                    complexVariantSVIntervalTree.overlappers(svInterval)
+                            .forEachRemaining(variantContextEntry -> stringBuilder.append(variantContextEntry.getValue().getID()).append(";"));
+                    targetContainerCoveredByComplexVariants.put(invBreakEndContext, stringBuilder.toString());
+                } else {
+                    targetContainerPicked.put(svInterval, invBreakEndContext);
+                }
+            }
+        }
+    }
+
+    //==================================================================================================================
+
+    /**
+     * Further cuts down the cases to look at after applying filters in {@link PrimitiveFilteringResult},
+     * that is, "filter" out breakends that
+     *   1) is not paired, i.e. the interval bounded by the suggested novel adjacency does not overlap with any other intra-chromosome SS BND s;
+     *   2) is paired with multiple intra-chromosome SS BND s;
+     *   3) is paired with one intra-chromosome SS BND, but a INV55/INV55 or INV33/INV33 pair
+     */
+    private static final class OverlapBasedFilteringResult {
+        private final Set<InvBreakEndContext> noOverlappers;                                        // variants whose has no overlapping INV BND suspects, or whose unique overlapper is one of {@code multipleOverlappers}
+        private final Map<InvBreakEndContext, List<String>> multipleOverlappers;                    // variants who overlaps multiple other records (note: not using map because VC.equals() is hard)
+        private final Set<Tuple2<InvBreakEndContext, InvBreakEndContext>> uniqueStrangeOverlappers; // pairs of variants (INV55/55, INV33/33) that overlap each other uniquely
+
+        private final Set<OverlappingPair> uniqueHetOverlappers;                                    // pairs of variants (INV55/33, INV33/55) that overlap each other uniquely, pairs that will be analyzed
+
+        OverlapBasedFilteringResult(final Set<InvBreakEndContext> noOverlappers,
```

Comment:
Have you thought about storing these breakpoints in a graph structure?

Then you could traverse nodes in the graph and get their category on the fly instead of splitting them into multiple collections.